server/src/main/java/org/candlepin/controller/ImportResult.java

```diff
@@ -0,0 +1,126 @@
+ * controller-level import operation.
+ *
+ * @param <E>
+ *  The entity class contained by a given import result instance
+ */
+public class ImportResult<E extends Persisted> {
+
+    private final Map<String, E> skippedEntities;
+    private final Map<String, E> createdEntities;
+    private final Map<String, E> updatedEntities;
+    private Map<String, E> importedEntities;
+
+    /**
+     * Instantiates a new ImportResult instance, using the given maps as its result set. Changes
+     * made to the maps after creating an import result will be reflected by this object.
+     *
+     * @param skippedEntities
+     *  A mapping of entities which were skipped during the import
+     *
+     * @param createdEntities
+     *  A mapping of entities which were created during the import
+     *
+     * @param updatedEntities
+     *  A mapping of entities which were updated during the import
+     *
+     * @throws IllegalArgumentException
+     *  if any of the provided maps are null
+     */
+    public ImportResult(Map<String, E> skippedEntities, Map<String, E> createdEntities,
+        Map<String, E> updatedEntities) {
+
+        if (skippedEntities == null) {
+            throw new IllegalArgumentException("skippedEntities is null");
+        }
+
+        if (skippedEntities == null) {
+            throw new IllegalArgumentException("skippedEntities is null");
+        }
+
+        if (skippedEntities == null) {
+            throw new IllegalArgumentException("skippedEntities is null");
+        }
+
+        this.skippedEntities = skippedEntities;
+        this.createdEntities = createdEntities;
+        this.updatedEntities = updatedEntities;
+        this.importedEntities = null;
+    }
+
+    /**
+     * Retrieves a map containing the skipped entities. The entities will be mapped by their Red
+     * Hat ID.
+     *
+     * @return
+     *  A map containing all skipped entities
+     */
+    public Map<String, E> getSkippedEntities() {
+        return this.skippedEntities;
+    }
+
+    /**
+     * Retrieves a map containing the created entities. The entities will be mapped by their Red
+     * Hat ID.
+     *
+     * @return
+     *  A map containing all created entities
+     */
+    public Map<String, E> getCreatedEntities() {
+        return this.createdEntities;
+    }
+
+    /**
+     * Retrieves a map containing the updated entities. The entities will be mapped by their Red
+     * Hat ID.
+     *
+     * @return
+     *  A map containing all updated entities
+     */
+    public Map<String, E> getUpdatedEntities() {
+        return this.updatedEntities;
+    }
+
+    /**
+     * Retrieves a map containing the imported entities. The entities will be mapped by their Red
+     * Hat ID.
+     *
+     * @return
+     *  A map containing all imported entities
+     */
+    public Map<String, E> getImportedEntities() {
+        if (this.importedEntities == null) {
```

Comment:
We'll still be doing this coalescing at some point (see the code removed from the end of the original importProducts/Content), this method just turns it into a lazy version of it.